src/pybind/mgr/snap_schedule/module.py

```diff
@@ -0,0 +1,178 @@
+
+LGPL2.1.  See file COPYING.
+"""
+import errno
+import json
+import sqlite3
+from .fs.schedule_client import SnapSchedClient
+from mgr_module import MgrModule, CLIReadCommand, CLIWriteCommand
+from mgr_util import CephfsConnectionException
+from threading import Event
+
+
+class Module(MgrModule):
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self._initialized = Event()
+        self.client = SnapSchedClient(self)
+
+    def resolve_subvolume_path(self, fs, subvol, path):
+        if not subvol:
+            return path
+
+        rc, subvol_path, err = self.remote('fs', 'subvolume', 'getpath',
+                                           fs, subvol)
+        if rc != 0:
+            # TODO custom exception?
+            raise Exception(f'Could not resolve {path} in {fs}, {subvol}')
+        return subvol_path + path
+
+    @property
+    def default_fs(self):
+        fs_map = self.get('fs_map')
+        if fs_map['filesystems']:
+            return fs_map['filesystems'][0]['mdsmap']['fs_name']
+        else:
+            self.log.error('No filesystem instance could be found.')
+            raise CephfsConnectionException(
+                -errno.ENOENT, "no filesystem found")
+
+    def serve(self):
+        self._initialized.set()
+
+    def handle_command(self, inbuf, cmd):
+        self._initialized.wait()
+        return -errno.EINVAL, "", "Unknown command"
+
+    @CLIReadCommand('fs snap-schedule status',
+                    'name=path,type=CephString,req=false '
+                    'name=subvol,type=CephString,req=false '
+                    'name=fs,type=CephString,req=false',
+                    'List current snapshot schedules')
+    def snap_schedule_get(self, path='/', subvol=None, fs=None):
+        use_fs = fs if fs else self.default_fs
+        try:
+            ret_scheds = self.client.get_snap_schedules(use_fs, path)
+        except CephfsConnectionException as e:
+            return e.to_tuple()
+        return 0, '\n===\n'.join([ret_sched.report() for ret_sched in ret_scheds]), ''
+
+    @CLIReadCommand('fs snap-schedule list',
+                    'name=path,type=CephString '
+                    'name=recursive,type=CephString,req=false '
+                    'name=subvol,type=CephString,req=false '
+                    'name=fs,type=CephString,req=false',
+                    'Get current snapshot schedule for <path>')
+    def snap_schedule_list(self, path, subvol=None, recursive=False, fs=None):
+        try:
+            use_fs = fs if fs else self.default_fs
+            scheds = self.client.list_snap_schedules(use_fs, path, recursive)
+            self.log.debug(f'recursive is {recursive}')
+        except CephfsConnectionException as e:
+            return e.to_tuple()
+        if not scheds:
+            return errno.ENOENT, '', f'SnapSchedule for {path} not found'
+        return 0, '\n'.join([str(sched) for sched in scheds]), ''
```

Comment:
Any reason this is not a json format like it was earlier? (the tests rely on output to be json)

Reply:
I anticipate the `list` command being used by a syncing tool, to find out where to check for new snapshots to be synced. So the idea is to have output here that can be easily parsed with bash tools.

Probably good the add a `--format json` option.